internal/pkg/source/github/repo_auth_flow.go

```
@@ -28,9 +31,16 @@ func PerformGithubAuthorization(cliContext runtime.CliContext) bool {
 	consoleWriter := cliContext.Out()
 	getEnvConfig := config.CreateConfigReader(cliContext.EnvConfig(), client.EnvConfigs)
 
+	state := obtainState(cliContext)
+	if state == "" {
+		common.ExitWithErrorMessage(consoleWriter, "Error while initiating authorization flow")
```

We will get two errors each time for an error. Ideally, we should throw an error in lower functions and print only in higher methods.

---

https://github.com/wso2/choreo-cli/pull/41#discussion_r333499992

---

This is different that the mentioned comment.

---

1st error will be printed when verbose info is needed. 2nd error is what a typical user will see

---

changed